Mathlib/Order/KrullDimension.lean

```diff
@@ -251,6 +251,73 @@ lemma height_le_coe_iff (x : α) (n : ℕ) :
   · simp
   · norm_cast
 
+/--
+The height of an element is infinite if there exist series of arbitrary length ending in that
+element.
+-/
+lemma height_eq_top_iff (x : α) :
+    height x = ⊤ ↔ ∀ n, ∃ p : LTSeries α, p.last = x ∧ p.length = n := by
+  constructor
+  · intro h n
+    apply exists_series_of_le_height x (n := n)
+    simp [h]
+  · intro h
+    rw [height_eq_iSup_last_eq, iSup_subtype', ENat.iSup_coe_eq_top, bddAbove_def]
+    push_neg
+    intro n
+    obtain ⟨p, hlast, hp⟩ := h (n+1)
+    exact ⟨p.length, ⟨⟨⟨p, hlast⟩, by simp [hp]⟩, by simp [hp]⟩⟩
+
+@[simp] lemma height_eq_zero (x : α) : height x = 0 ↔ IsMin x := by
+  simpa [isMin_iff_forall_not_lt] using height_le_coe_iff x 0
+
+protected alias ⟨_, IsMin.height_eq_zero⟩ := height_eq_zero
+
+@[simp] lemma height_bot (α : Type*) [Preorder α] [OrderBot α] : height (⊥ : α) = 0 := by simp
+
+lemma coe_lt_height_iff (x : α) (n : ℕ) (hfin : height x < ⊤) :
+    n < height x ↔ (∃ y, y < x ∧ height y = n) where
+  mp h := by
+    obtain ⟨m, hx : height x = m⟩ := Option.ne_none_iff_exists'.mp (LT.lt.ne_top hfin)
+    rw [hx] at h; norm_cast at h
+    obtain ⟨p, hp, hlen⟩ := exists_series_of_height_eq_coe x hx
+    use p ⟨n, by omega⟩
+    constructor
+    · rw [← hp]
+      apply LTSeries.strictMono
+      simp [Fin.last]; omega
+    · exact height_eq_index_of_length_eq_height_last (by simp [hlen, hp, hx]) ⟨n, by omega⟩
+  mpr := fun ⟨y, hyx, hy⟩ =>
+    hy ▸ height_strictMono hyx (lt_of_le_of_lt (height_mono hyx.le) hfin)
+
+lemma height_eq_coe_add_one_iff (x : α) (n : ℕ) : height x = n + 1 ↔
+    height x < ⊤ ∧ (∃ y < x, height y = n) ∧ (∀ y, y < x → height y ≤ n) := by
```

Comment:
Can you avoid starting newlines in the middle of an inlined statement?
```suggestion
lemma height_eq_coe_add_one_iff (x : α) (n : ℕ) :
    height x = n + 1 ↔ height x < ⊤ ∧ (∃ y < x, height y = n) ∧ (∀ y < x, height y ≤ n) := by
```
same below